### PR TITLE
Added the ability to set outside proxy url (close #1032)

### LINKED ISSUE #1032: Add the ability to set outside proxy url

### PR DESCRIPTION
**Pull request by LavrovArtem**
/cc @inikulin @churkin

### CODE REVIEW

File: src/messages.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 export const MESSAGE = {
     destConnectionTerminated: 'Failed to perform a request for the resource at <a href="{url}">{url}</a> because connection was unexpectedly terminated.',
     cantResolveUrl:           'Failed to find a DNS-record for the resource at <a href="{url}">{url}</a>.',
+    cantTunnelingConnection:  'Failed to perform a tunneling connection for the host at <a href="{url}">{url}</a>.',
```
Comment (VasilyStrelyaev) [inline src/messages.js]:
```
    cantEstablishTunnelingConnection:  'Failed to establish tunneling connection to the host at <a href="{url}">{url}</a>.',
```

**Comment by LavrovArtem** [comment]
FPR

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
FPR

**Comment by LavrovArtem** [comment]
ping @inikulin @churkin

File: src/messages.js [hunk 2]
```diff
@@ -1,7 +1,8 @@
 export const MESSAGE = {
-    destConnectionTerminated: 'Failed to perform a request for the resource at <a href="{url}">{url}</a> because connection was unexpectedly terminated.',
-    cantResolveUrl:           'Failed to find a DNS-record for the resource at <a href="{url}">{url}</a>.',
-    destRequestTimeout:       'Failed to complete a request to <a href="{url}">{url}</a> within the timeout period. The problem may be related to local machine\'s network or firewall settings, server outage, or network problems that make the server inaccessible.'
+    destConnectionTerminated:         'Failed to perform a request for the resource at <a href="{url}">{url}</a> because connection was unexpectedly terminated.',
+    cantResolveUrl:                   'Failed to find a DNS-record for the resource at <a href="{url}">{url}</a>.',
+    cantEstablishTunnelingConnection: 'Failed to establish tunneling connection to the host at <a href="{url}">{url}</a>.',
```
Comment (inikulin) [inline src/messages.js]: We need to highlight that it's connection to the proxy

File: src/proxy/index.js [hunk 3]
```diff
@@ -130,12 +130,12 @@ export default class Proxy extends Router {
         this._closeSockets();
     }
 
-    openSession (url, session, externalProxySettings) {
+    openSession (url, session, proxyUrl) {
```
Comment (inikulin) [inline src/proxy/index.js]: name `externalProxyUrl` not confuse it with hammerhead

File: src/proxy/index.js [hunk 4]
```diff
@@ -130,12 +130,12 @@ export default class Proxy extends Router {
...
         session.proxy                 = this;
         this.openSessions[session.id] = session;
 
-        if (externalProxySettings)
-            session.setExternalProxySettings(externalProxySettings.url, externalProxySettings.ignoreHosts);
+        if (proxyUrl !== void 0)
```
Comment (inikulin) [inline src/proxy/index.js]: `if(proxyUrl)`

File: src/request-pipeline/destination-request/index.js [hunk 5]
```diff
@@ -113,6 +119,9 @@ export default class DestinationRequest extends EventEmitter {
             this._send();
         }
 
+        else if (this.isHttps && this.opts.proxy && DestinationRequest._isTunnelingErr(err))
```
Comment (inikulin) [inline src/request-pipeline/destination-request/index.js]: Just move whole condition to `_isExternalProxyTunelingErr` method

File: src/messages.js (hunk 2 above)
Comment (VasilyStrelyaev) [inline src/messages.js]: Failed to connect to the proxy. Cannot establish tunneling connection to the host at <a href="{url}">{url}</a>.

**Comment by LavrovArtem** [comment]
FPR

**Comment by LavrovArtem** [comment]
ping @inikulin @churkin

**Pull request merged**